components/config/opts.rs

```diff
@@ -665,16 +664,12 @@ pub fn from_cmdline_args(args: &[String]) -> ArgumentParsingResult {
              url.starts_with("http://web-platform.test:8000/_mozilla/mozilla/canvas/") ||
              url.starts_with("http://web-platform.test:8000/_mozilla/css/canvas_over_area.html"));
 
-    let url = match url_opt {
-        Some(url_string) => {
-            parse_url_or_filename(&cwd, url_string)
-                .unwrap_or_else(|()| args_fail("URL parsing failed"))
-        },
-        None => {
-            print_usage(app_name, &opts);
-            args_fail("servo asks that you provide a URL")
-        }
-    };
+    let url_opt = url_opt.and_then(|url_string| parse_url_or_filename(&cwd, url_string).ok()
+                                   // If url parsing fails, print error and assign None
+                                   .or_else(|| {
+                                       println!("URL parsing failed.");
+                                       None
+                                   }));
```

Comment:
You should use `warn!` instead of `println!`.

Also, `ok` transformed the `Result` into an `Option`. A `Result` is basically like an option, but with an error message instead of `None`.

So since we print a warning if it fails to parse, maybe we also want to display the error message that comes with `parse_url_or_filename`.

```rust
parse_url_or_filename(&cwd, url_string).or_else(|error| {
 warn!("URL parsing failed ({})", error);
 error
}).ok()
```

Does it make sense?